prov/verbs/src/fi_verbs.c

```diff
@@ -612,58 +610,71 @@ fi_ibv_getepinfo(const char *node, const char *service,
 			return ret;
 
 		ret = rdma_getaddrinfo((char *) node, (char *) service,
-					&rai_hints, &rai);
+					&rai_hints, rai);
 	} else {
 		ret = rdma_getaddrinfo((char *) node, (char *) service,
-					NULL, &rai);
-	}
-	if (ret)
-		return (errno == ENODEV) ? -FI_ENODATA : -errno;
-
-	if (!(fi = fi_allocinfo_internal())) {
-		ret = -FI_ENOMEM;
-		goto err1;
-	}
-
-	ret = fi_ibv_rai_to_fi(rai, hints, fi);
-	if (ret)
-		goto err2;
-
-	ret = rdma_create_ep(id, rai, NULL, NULL);
-	if (ret) {
-		ret = -errno;
-		goto err2;
+					NULL, rai);
 	}
 
-	ret = fi_ibv_fill_info_attr((*id)->verbs, hints, fi);
 	if (ret)
-		goto err3;
+		return (errno == ENODEV) ? -FI_ENODATA : -errno;
 
-	*info = fi;
-	rdma_freeaddrinfo(rai);
 	return 0;
-
-err3:
-	rdma_destroy_ep(*id);
-err2:
-	fi_freeinfo(fi);
-err1:
-	rdma_freeaddrinfo(rai);
-	return ret;
 }
 
 static int fi_ibv_getinfo(uint32_t version, const char *node, const char *service,
 			  uint64_t flags, struct fi_info *hints, struct fi_info **info)
 {
+	struct rdma_addrinfo *rai, *rhead;
+	struct fi_info *fi, *fhead = NULL;
+	int ret, have_head = 0;
 	struct rdma_cm_id *id;
-	int ret;
 
-	ret = fi_ibv_getepinfo(node, service, flags, hints, info, &id);
+	ret = fi_ibv_getepinfo(node, service, flags, hints, &rai);
+
 	if (ret)
 		return ret;
 
-	rdma_destroy_ep(id);
-	return 0;
+	rhead = rai;
+
+	for (struct fi_info *tmp; rai; rai = rai->ai_next) {
```

Comment:
Minor stylistic comment -- you can ignore if you don't like it: you might want to set rhead=rai as part of the loop.  E.g:

```c
for (struct fi_info *tmp, rhead = rai; rai; rai = rai->ai_next)
  ...
```